### ISSUE TITLE
How do you have gt amodal segmentation during test phase

### ISSUE THREAD

**Issue by DonaldKam** [YihongSun/Bayesian-Amodal#6]
Hi, how do you have gt amodal segmentation during test phase? Are the images with occlusion synthetic or inferred manually?

**Comment by YihongSun** [comment]
Hello, 

1. gt amodal segmentation is predicted at inference time, not given. I can clarify further if there is a specific line that you can point to. 

2. Please refer to Section 4.1 and Figure 4 of the [manuscript](https://arxiv.org/pdf/2010.13175.pdf), _KINS_ and _COCOA-cls_ contain real occlusion, while _OccludedVehicles_ contains synthetic occlusion.

Feel free to reply if there is anything else that I can help with!

**Comment by DonaldKam** [comment]
Thank you. Regarding the E2E one, do you train the model for object detection and amodal segmentation meanwhile? If so, amodal segmentation would be performed upon object detection, which seems difficult as the two tasks are not parallel (segmentation is within the bounding box), right? Another question, from the test results, why only one vehicle is segmented in each demo image? In the paper there are multiple segmented vehicles in each image.

**Comment by YihongSun** [comment]
Thanks for your questions.

1. Please refer to Section 3.1 of the [manuscript](https://arxiv.org/pdf/2010.13175.pdf). Our work assumes a given bounding box as input, which means that amodal segmentation is performed on top of the detection results. In Section 4.1, we detail our evaluation with using the best region proposal from a RPN as input for the respective approaches.
2. Stemming from the previous answer, amodal segmentation is predicted for each given box. The number of segmentation would depend on the number of given bounding boxes.

**Comment by DonaldKam** [comment]
Thanks for prompt reply. 'In Section 4.1, we detail our evaluation with using the best region proposal from a RPN as input for the respective approaches'-----Do you mean that you integrate the detection (RPN) and the segmention as one network? If yes, why not use RPN during training phase? If not, why not use an single detector that outputs better quality Bboxes than thoes from RPN?

**Comment by YihongSun** [comment]
1. The RPN predictions are incorporated during evaluation to simulate a more realistic setting where the given visible bounding box is not perfect. It is not considered a part of the method, as it is one of many options that supplies region proposals.
2. Our proposed method is on the task of learning amodal segmentation from visible bounding box supervision only, learning a detector that predicts better quality bounding boxes is not within the scope of the paper.

**Comment by DonaldKam** [comment]
Thanks a lot. When I plotted the bounding boxes on KINS, I found that the coordinates didn't correspond to the objects. How to understand the amodal_bbox.npy file? Are they left top x, left y, width, height?![mmexport1694168252569.png](https://github.com/YihongSun/Bayesian-Amodal/assets/44252954/c735a16d-fa99-40e1-bec9-168fa46416fb)

**Comment by YihongSun** [comment]
Hi,
Judging based on the KITTI image dimension (1242 x 375), I would assume the coordinates are [left, top, right, bottom]. Regardless, I would suggest comparing against the acutal image and see how the boxes are used in the code to verify.

**Comment by DonaldKam** [comment]
> Thanks for your questions.
> 
> 1. Please refer to Section 3.1 of the [manuscript](https://arxiv.org/pdf/2010.13175.pdf). Our work assumes a given bounding box as input, which means that amodal segmentation is performed on top of the detection results. In Section 4.1, we detail our evaluation with using the best region proposal from a RPN as input for the respective approaches.
> 2. Stemming from the previous answer, amodal segmentation is predicted for each given box. The number of segmentation would depend on the number of given bounding boxes.

Hi. Why don't you use RPN for both training phase and test (evaluation) phase? Then it will be a completely e2e method, from detection to segmentation.

**Comment by YihongSun** [comment]
The RPN is trained separately and its predictions are not incorporated into the training of our architecture. It would be interesting to see how that would affect learning, but since the architecture is first learned with EM, my guess would be that the regions predicted by RPN may be too noise to properly learn the Bayesian model.

**Comment by DonaldKam** [comment]
Thank you for prompt reply. I imported the KINS annotation files and found the amodal_mask data was not stored in coordinates like COCO. Do you know how to encode the coordinates of mask into the KINS format?
![267690318-ded8f6cf-5f3e-4db0-ad68-d553079e7b11](https://github.com/YihongSun/Bayesian-Amodal/assets/44252954/03eea70e-a538-4191-b932-988e489091ca)

**Comment by DonaldKam** [comment]
When applying the method in customized dataset, how to train the backbone to acquire RPN bboxes?

**Comment by YihongSun** [comment]
Since the RPN training is disjoint, we utilize the standard training scheme and RPN architecture found in [Mask R-CNN](https://pytorch.org/vision/main/models/generated/torchvision.models.detection.maskrcnn_resnet50_fpn.html#torchvision.models.detection.maskrcnn_resnet50_fpn).

**Comment by DonaldKam** [comment]
One more question, could you show me how the known object center _c_ contributes to the amodal prediction? Concretely, how is _c_ processed during modeling?

**Comment by FedericaGielen** [comment]
> One more question, could you show me how the known object center _c_ contributes to the amodal prediction? Concretely, how is _c_ processed during modeling?

Same question here

**Comment by YihongSun** [comment]
In the paper, we perform displacement modeling (Sec 3.3) that allow us to align the partial visible bounding box to our object representation.

In order to fairly compare with related works that only receive amodal/complete bounding box during training (e.g. BBTP), we consider the case where the true object center is known, which allows us to explicitly skip the displacement modeling step.

In practice, we select the region proposal with the highest IoU to the gt amodal/complete box for _known center c_, and the region proposal with the highest IoU to the gt partial/visible box for _unknown center c_.
